File: apps/mobile/android/app/src/main/java/com/visvine/mobile/ui/theme/ColorThemes.kt
package com.visvine.mobile.ui.theme

import androidx.compose.ui.graphics.Color

/**
 * The colours a screen draws with: the semantic design tokens ([VVColor], from
 * packages/tokens via Tokens.kt) plus the accent the person picked. The names
 * are the token roles — the same names the web's utilities (`text-fg-muted`,
 * `bg-surface-subtle`) and iOS's `DynamicColors` use.
 */
data class DynamicColors(
    /** The chosen hue: fills and the selected state. */
    val accent: Color,
    /** Text and icons in the accent. */
    val accentStrong: Color,
    /** A wash behind a selected row. */
    val accentSoft: Color,
    val surface: Color = VVColor.surface,
    val surfaceSubtle: Color = VVColor.surfaceSubtle,
    val surfaceMuted: Color = VVColor.surfaceMuted,
    val surfaceGlass: Color = VVColor.surfaceGlass,
    val fg: Color = VVColor.fg,
    val fgSecondary: Color = VVColor.fgSecondary,
    val fgMuted: Color = VVColor.fgMuted,
    val fgSubtle: Color = VVColor.fgSubtle,
    val line: Color = VVColor.line,
    val lineSubtle: Color = VVColor.lineSubtle,
    val danger: Color = VVColor.danger,
    val success: Color = VVColor.success,
    val warning: Color = VVColor.warning,
)

/** The palette for one accent. The app is light-only, so this is one palette, not two. */
fun buildColors(accent: VVAccent): DynamicColors =
    DynamicColors(accent = accent.light.base, accentStrong = accent.light.strong, accentSoft = accent.light.soft)

File: apps/mobile/ios/Visvine/Theme/ColorThemes.swift
import SwiftUI

/// The colours a view draws with: the semantic design tokens
/// (`Tokens.generated.swift`, from packages/tokens) plus the accent the person
/// picked. The names are the token roles — the same names the web's utilities
/// (`text-fg-muted`, `bg-surface-subtle`) and Android's `VVColor` use.
struct DynamicColors {
    /// The chosen hue: fills and the selected state.
    let accent: Color
    /// Text and icons in the accent.
    let accentStrong: Color
    /// A wash behind a selected row.
    let accentSoft: Color

    let surface = VVColor.surface
    let surfaceSubtle = VVColor.surfaceSubtle
    let surfaceMuted = VVColor.surfaceMuted
    let fg = VVColor.fg
    let fgSecondary = VVColor.fgSecondary
    let fgMuted = VVColor.fgMuted
    let fgSubtle = VVColor.fgSubtle
    let line = VVColor.line
    let lineSubtle = VVColor.lineSubtle
    let danger = VVColor.danger
    let success = VVColor.success
    let warning = VVColor.warning
}

/// The palette for one accent. The app is light-only, so this is one palette, not two.
func buildColors(accent: VVAccent) -> DynamicColors {
    DynamicColors(accent: accent.base, accentStrong: accent.strong, accentSoft: accent.soft)
}
